lib/components/app/responsive-webapp.js

```diff
@@ -102,17 +104,96 @@ class ResponsiveWebapp extends Component {
       )
     }
 
-    // Check for initial display state
-    if (initialDisplay === 'route' && location.search) {
-      const params = qs.parse(location.search.substring(1))
-      this.props.setMainPanelContent(MainPanelContent.ROUTE_VIEWER)
-      if (params.id) {
-        this.props.findRoute({ routeId: params.id })
-        this.props.setViewedRoute({ routeId: params.id })
+    if (location && location.search) {
+      // Set search params and plan trip if routing enabled and a query exists
+      // in the URL.
+      this.props.parseUrlQueryString()
+    }
+    // Handle routing to a specific part of the app (e.g. stop viewer) on page
+    // load.
+    this._matchContentToUrl(this.props)
+  }
+
+  componentWillUnmount () {
+    // Remove on back button press listener.
+    window.onpopstate = () => {}
+  }
+
+  _onBackButtonPress = (e) => {
+    // Get new search ID from URL after back button pressed.
+    // console.log('back button pressed', e)
+    const urlParams = getUrlParams()
+    const newSearchId = urlParams.ui_activeSearch
+    const newItinIndex = +urlParams.ui_activeItinerary || 0
+    const newSearch = this.props.searches[newSearchId]
+    if (newSearch) {
```

Comment:
Would it make more sense to rename these as `previous...` instead of `new...`?

Reply:
I see what you mean there, but on second thought, this function is also called when the 'forward' button is pressed, so previous might not always be an appropriate term.

Reply:
I went ahead and just changed to previous.